src/cwe_checker_lib/src/intermediate_representation/expression.rs

```diff
@@ -431,5 +432,79 @@ impl Expression {
     }
 }
 
+impl fmt::Display for Expression {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Expression::Var(var) => write!(f, "{}", var),
+            Expression::Const(c) => {
+                write!(f, "0x{:016x}:i{}", c, c.bytesize().as_bit_length())
+            }
+            Expression::BinOp { op, lhs, rhs } => match op {
+                BinOpType::IntMult
+                | BinOpType::IntDiv
+                | BinOpType::IntRem
+                | BinOpType::FloatMult
+                | BinOpType::FloatDiv => write!(f, "{} {} {}", lhs, op, rhs),
+                _ => write!(f, "({} {} {})", lhs, op, rhs),
+            },
+            Expression::UnOp { op, arg } => write!(f, "{}({})", op, arg),
+            Expression::Cast { op, size: _, arg } => write!(f, "{}({})", op, arg),
+            Expression::Unknown {
+                description,
+                size: _,
+            } => write!(f, "{}", description),
+            Expression::Subpiece {
+                low_byte,
+                size,
+                arg,
+            } => {
+                if let (Ok(start), Ok(end)) = (u32::try_from(low_byte.0), u32::try_from(size.0)) {
+                    write!(f, "({})[{}₋{}]", arg, start, end)
```

Comment:
For some reason the character you use as hyphen `₋` is not the normal `-` (minus) character. I think we should use the normal `-`.